var/spack/repos/builtin/packages/umpire/package.py

```diff
@@ -104,9 +104,8 @@ class Umpire(CachedCMakePackage, CudaPackage, ROCmPackage):
 
     # https://github.com/LLNL/Umpire/issues/653
     # This range looks weird, but it ensures the concretizer looks at it as a
-    # range, not as a concrete version, so that it also matches compilers
-    # specified as `gcc@10.3.0-identifier`. See #8957.
-    conflicts('%gcc@10.3.0:10.3.0.0', when='+cuda')
+    # range, not as a concrete version, so that it also matches 10.3.* versions.
+    conflicts('%gcc@10.3.0:10.4.0', when='+cuda')
```

Comment:
```suggestion
    conflicts('%gcc@10.3.0:10.3', when='+cuda')
```
If you know for a fact they're not going to backport to 10.4, then it's fine. If you want >= 10.3.0 and < 10.4 then the suggestion is the way to go

Reply:
It's a good point, sorry for that.

According to this comment from the GCC bug tracker https://gcc.gnu.org/bugzilla/show_bug.cgi?id=100102#c25
> Patrick Palka 2021-06-07 22:56:00 UTC
> Fixed for GCC 10.4, 11.2 and 12.

while according to the patch we hope it fixes this problem, it is slightly different.
https://github.com/spack/spack/blob/a5bd6acd9061312aea58f180da59250d880b9a26/var/spack/repos/builtin/packages/gcc/package.py#L298-L300

what do you suggest?

Reply:
I think we should target system compilers that don't have the patch, so `10.3.0:10.3` is fine then...

Reply:
Also 11.0 and 11.1 should be addressed in theory...